compiler/ballerina-lang/src/main/java/org/ballerinalang/model/clauses/JoinClauseNode.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.model.clauses;
+
+import org.ballerinalang.model.tree.Node;
+import org.ballerinalang.model.tree.expressions.ExpressionNode;
+import org.ballerinalang.model.tree.statements.VariableDefinitionNode;
+
+/**
+ * The interface with the APIs to implement the "join" clause.
+ *
+ * @since 1.3.0
+ */
+public interface JoinClauseNode extends Node {
+
+    ExpressionNode getCollection();
+
+    void setCollection(ExpressionNode collection);
+
+    boolean setDeclaredWithVar();
+
+    boolean isDeclaredWithVar();
+
+    VariableDefinitionNode getVariableDefinitionNode();
+
+    void setVariableDefinitionNode(VariableDefinitionNode variableDefinitionNode);
+
+    OnClauseNode getOnClauseNode();
```

Comment:
Why we keep OnClauseNode here? It's also a top-level node, right? Similar to where clause.